src/source/rtl_text_plugin.js

```diff
@@ -1,11 +1,30 @@
 // @flow
 
 import { Event, Evented } from '../util/evented';
+import window from '../util/window';
 
 let pluginRequested = false;
 let pluginURL = null;
 let foregroundLoadComplete = false;
 
+const isAbsoluteUrl = (url) => {
+    // handles https, http, and protocol-relative URLs
+    const isAbsolute = new RegExp('^([a-z]+://|//)', 'i');
+    return isAbsolute.exec(url);
+};
+
+const resolvePluginUrl = (path) => {
+    if (isAbsoluteUrl(path)) {
+        return path;
+    }
+    // resolves from root URL
+    // if path begins with a '/', it will be appended to the domain
+    // otherwise, it will be appended to the full path
+    const a = window.document.createElement('a');
+    a.href = path;
+    return a.href;
+};
```

Comment:
Looks like we're doing something similar in https://github.com/mapbox/mapbox-gl-js/blob/6255e9e449d1d2cffb9810de57e9a8d97275a500/src/source/geojson_source.js#L288-L292

Can we move that code to a more central place to avoid duplicating it?

Reply:
Ah thanks for pointing that out. Yeah, we should be able to centralize this somewhere.